core/builder/src/main/java/io/quarkus/builder/Json.java

```diff
@@ -311,25 +409,67 @@ protected JsonObjectBuilder self() {
             return this;
         }
 
+        @Override
+        void with(JsonReader.JsonValue element) {
+            if (element instanceof JsonReader.JsonMember) {
+                final JsonReader.JsonMember member = (JsonReader.JsonMember) element;
+                final String attribute = member.attribute().value();
+                final JsonReader.JsonValue value = member.value();
+                if (value instanceof JsonReader.JsonString) {
+                    put(attribute, ((JsonReader.JsonString) value).value());
```

Comment:
I looked into this and I'm not sure about it. The object builder methods `put` calls take an attribute and value there, whereas the array builder ones only take a value, so I don't think it's a good idea to bunch them all together at object builder class. The duplication could be abstracted away in other ways but I'm unsure there's a real tangible benefit since the JSON grammar is set and we're not going to evolve this further.